File: backend/Models/CourseEnrollment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VideoIntelligencePlatform.Backend.Models;

[Table("course_enrollments")]
public class CourseEnrollment
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("course_id")]
    public int CourseId { get; set; }

    [Column("user_id")]
    public int UserId { get; set; }

    [Column("enrolled_at")]
    public DateTime EnrolledAt { get; set; } = DateTime.UtcNow;

    [Column("amount_paid")]
    public decimal AmountPaid { get; set; } = 0.0m;

    [ForeignKey("CourseId")]
    public Course? Course { get; set; }

    [ForeignKey("UserId")]
    public User? User { get; set; }
}

File: backend/app/models/course_enrollment.py
from datetime import datetime
from sqlalchemy import Column, Integer, Numeric, DateTime, ForeignKey, UniqueConstraint
from sqlalchemy.orm import relationship
from app.database import Base


class CourseEnrollment(Base):
    __tablename__ = "course_enrollments"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    course_id = Column(Integer, ForeignKey("courses.id", ondelete="CASCADE"), nullable=False)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    enrolled_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    amount_paid = Column(Numeric(10, 2), nullable=False, default=0.0)

    __table_args__ = (
        UniqueConstraint("course_id", "user_id", name="uq_course_enrollment"),
    )

    # Relationships
    course = relationship("Course", back_populates="enrollments")
    user = relationship("User", back_populates="enrollments")
